puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tfmROR52.groovy

```diff
@@ -0,0 +1,67 @@
+def packages_to_build
+
+pipeline {
+    agent { label 'rpmbuild' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '15'))
+        disableConcurrentBuilds()
+    }
+
+    stages {
+        stage('Clone Packaging') {
+            steps {
+
+                deleteDir()
+
+                checkout([
+                    $class : 'GitSCM',
+                    branches : [[name: '*/tfm-ror52']],
+                    userRemoteConfigs: [
+                        [url: 'https://github.com/theforeman/rails-packaging']
+                    ]
+                ])
+
+            }
+        }
+        stage('Find packages') {
+            steps {
+
+                copyArtifacts(projectName: env.JOB_NAME, optional: true)
+
+                script {
+
+                    if (fileExists('commit')) {
+                        commit = readFile(file: 'commit').trim()
+
+                        if (commit) {
+                            packages_to_build = find_changed_packages("${commit}..HEAD")
+                        }
+                    } else {
+                        packages_to_build = 'all'
+                    }
+                }
+            }
+        }
+        stage('Release Build Packages') {
+            steps {
+
+                obal(
+                    action: 'release',
+                    packages: packages_to_build
+                )
+
+            }
+        }
+    }
+
+    post {
+        success {
+            sh "git rev-parse HEAD > commit"
```

Comment:
so the idea here is that after we run a build, we write the sha1 of `HEAD` to `commit` and archive that file, and the next run will fetch it again via `copyArtifacts` and know what changed? deserves a comment ;)

Reply:
That's the idea, can comment